server/routes/authRoutes.js

```diff
@@ -1,299 +1,334 @@
 
-        let t = token.signToken(existingUser._id);
-        if(!t) throw Error('Invalid Credentials')
-        console.log('cookie:' + t),
-        res.cookie('access_token', t, { httpOnly: true }),
-        res.json({...existingUsers._doc, msg: 'success', cookie: t}),
-        
-        next()
-        } catch (e) {
-        console.log(e);
-        res.status(400).send('Email and Password do not match');
-    }
+    let t = token.signToken(existingUser._id)
+    if (!t) throw Error('Invalid Credentials')
+    console.log('cookie:' + t),
+      res.cookie('access_token', t, { httpOnly: true }),
+      res.json({ ...existingUsers._doc, msg: 'success', cookie: t }),
+      next()
+  } catch (e) {
+    console.log(e)
+    res.status(400).send('Email and Password do not match')
+  }
 })
 
 router.get('/logout', (req, res) => {
-    let t = delToken.signToken(123);
-    req.logout();
-    res.cookie('access_token', t, { httpOnly: true }),
-    res.json({msg: 'success', cookie: t});
-});
+  let t = delToken.signToken(123)
+  req.logout()
+  res.cookie('access_token', t, { httpOnly: true }), res.json({ msg: 'success', cookie: t })
+})
 
 router.get('/current_user', requireLogin, async (req, res) => {
   try {
     const existinguser = await userHelper.getUserById(req.user.sub)
-    if(!existinguser) throw Error('user does not exist')
+    if (!existinguser) throw Error('user does not exist')
     res.json(existinguser)
   } catch (e) {
-    res.status(400).json({msg: e.message})
+    res.status(400).json({ msg: e.message })
   }
-});
+})
 
 router.post('/addMailing', async (req, res) => {
-  let {email, firstName, lastName} = req.body;
+  let { email, firstName, lastName } = req.body
   try {
-    const existingEmail = await emailList.findOne({email});
+    const existingEmail = await emailList.findOne({ email })
     if (!existingEmail) {
-    await emailList.create({email, firstName, lastName})
+      await emailList.create({ email, firstName, lastName })
     }
-    res.send({ message: 'success'})
+    res.send({ message: 'success' })
   } catch (e) {
-    res.status(400).send({ message: 'please try again'})
+    res.status(400).send({ message: 'please try again' })
   }
-});
+})
 
 router.post('/test', async (req, res) => {
-  let {email, firstName, lastName} = req.body;
+  let { email, firstName, lastName } = req.body
   try {
-    await emailList.create({email, firstName, lastName})
+    await emailList.create({ email, firstName, lastName })
     console.log(keys.mongoURI)
-    res.send({ message: keys.mongoURI})
+    res.send({ message: keys.mongoURI })
   } catch (e) {
-    res.status(400).send({ message: 'please try again'})
+    res.status(400).send({ message: 'please try again' })
   }
-});
+})
 
-  
+router.get('/github', async (req, res) => {
+  // 1) use the code to get token from github
+  const { code } = req.query
+  let githubToken
+  {
+    const body = {
+      client_id: process.env.NEXT_PUBLIC_GITHUB_CLIENT_ID,
```

Comment:
can you share the updated env file with me?

Reply:
https://clear-fit.slack.com/archives/C05RV492CUS/p1695310079288929 yep shared here